src/Mcfedr/QueueManagerBundle/Controller/PubSubController.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mcfedr\QueueManagerBundle\Controller;
+
+use Mcfedr\QueueManagerBundle\Exception\UnrecoverableJobExceptionInterface;
+use Mcfedr\QueueManagerBundle\Queue\PubSubJob;
+use Mcfedr\QueueManagerBundle\Runner\JobExecutor;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+
+class PubSubController extends AbstractController
+{
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+    /**
+     * @var JobExecutor
+     */
+    private $jobExecutor;
+
+    public function __construct(SerializerInterface $serializer, JobExecutor $jobExecutor)
+    {
+        $this->serializer = $serializer;
+        $this->jobExecutor = $jobExecutor;
+    }
+
+    /**
+     * @Route("/pubsub", name="pubsub", methods={"POST"})
+     */
+    public function pubsub(Request $request)
+    {
+        $message = json_decode($request->getContent(), true);
```

Comment:
So you had good intensions - injected Serializer... and then decided to forget about it...

Reply:
The Google api has a function for basic validation of the token - https://developers.google.com/identity/sign-in/web/backend-auth#verify-the-integrity-of-the-id-token